src/python/pants/engine/internals/build_files.py

```diff
@@ -178,28 +178,53 @@ def setup_address_specs_filter(global_options: GlobalOptions) -> AddressSpecsFil
     )
 
 
-@rule
-async def addresses_from_address_specs(
-    address_specs: AddressSpecs,
-    build_file_options: BuildFileOptions,
-    specs_filter: AddressSpecsFilter,
-) -> Addresses:
-    matched_addresses: OrderedSet[Address] = OrderedSet()
-    filtering_disabled = address_specs.filter_by_global_options is False
-
-    # Resolve all `AddressLiteralSpec`s. Will error on invalid addresses.
-    literal_wrapped_targets = await MultiGet(
+@rule_helper
+async def _determine_literal_addresses_from_specs(specs: AddressSpecs) -> tuple[WrappedTarget, ...]:
+    literal_addresses = await MultiGet(
         Get(
-            WrappedTarget,
+            Address,
             AddressInput(
                 spec.path_component,
                 spec.target_component,
                 spec.generated_component,
                 spec.parameters,
             ),
         )
-        for spec in address_specs.literals
+        for spec in specs.literals
```

Comment:
Since the helper only consumes the literals, it could maybe just take the literals as its argument.